crates/simulate/src/state/mod.rs

```diff
@@ -1,249 +1,367 @@
-                        .into_iter()
-                        .map(|tx| self.execute_evm_tx(tx))
-                        .collect::<Result<Vec<_>, SimStateError>>()?;
-                    let world_results = updates
-                        .into_iter()
-                        .map(|update| self.execute_world_update(update))
-                        .collect::<Vec<_>>();
-                    if let Some(tag) = agent_update.tag {
-                        self.insert_into_update_results(
-                            tag,
-                            agent_update.address,
-                            SimUpdateResult::MultiBundle(true, evm_results, world_results),
-                        );
-                    }
-                } else if let Some(tag) = agent_update.tag {
-                    self.insert_into_update_results(
-                        tag,
-                        agent_update.address,
-                        SimUpdateResult::MultiBundle(false, sim_evm_results, vec![]),
-                    );
-                }
-            }
+pub fn build_revm_tx(caller: Address, ethers_tx: TypedTransaction) -> StateResult<TxEnv> {
+    let transact_to = match ethers_tx.to_addr() {
+        Some(to) => TransactTo::Call(EvmAddress::from(*to)),
+        None => unreachable!(),
+    };
+    let tx_data = EvmBytes::from(
+        ethers_tx
+            .data()
+            .ok_or(StateError::EvmMissingTxData(ethers_tx.clone()))?
+            .to_vec(),
+    );
+    let tx_gas_settings = GasSettings::default();
+    Ok(TxEnv {
+        caller: caller.into(),
+        gas_limit: tx_gas_settings.gas_limit,
+        gas_price: tx_gas_settings.gas_price.into(),
+        gas_priority_fee: tx_gas_settings.gas_priority_fee.map(|x| x.into()),
+        transact_to,
+        value: U256::zero().into(),
+        data: tx_data,
+        chain_id: None,
+        nonce: None,
+        access_list: Vec::new(),
+    })
+}
+
+#[derive(Debug, Clone)]
+pub enum EvmTxOutput {
+    Success(EvmBytes),
+    SuccessCreate(Address),
+    Halt(Halt),
+    Revert(EvmBytes),
+}
+
+impl EvmTxOutput {
+    pub fn parse_success(self) -> EvmBytes {
+        match self {
+            EvmTxOutput::Success(output) => output,
+            _ => panic!("EvmTxOutput is not success."),
         }
+    }
+}
 
-        Ok(())
+pub fn unpack_evm_tx_output(result: ExecutionResult) -> StateResult<EvmTxOutput> {
+    match result {
+        ExecutionResult::Halt { reason, .. } => Ok(EvmTxOutput::Halt(reason)),
+        ExecutionResult::Revert { output, .. } => Ok(EvmTxOutput::Revert(output)),
+        ExecutionResult::Success { output, .. } => match output {
+            Output::Call(output) => Ok(EvmTxOutput::Success(output)),
+            Output::Create(_, address) => Ok(EvmTxOutput::SuccessCreate(
+                address.ok_or(StateError::MissingCreateAddress)?.into(),
+            )),
+        },
+    }
+}
+
+#[derive(Debug)]
+pub struct StateMiddleware();
```

Comment:
This is a dummy ethers `Middleware` implementation, which does not really do anything. We need it to pass around ethers `ContractCall` across the agents, state and manager. The issue is basically that you need a `Middleware` in order to generate a contract instance directly from the rust bindings.